subsys/bluetooth/host/hci_core.c

```diff
@@ -577,6 +579,7 @@ static int set_le_adv_enable_ext(const struct bt_le_ext_adv *adv,
 		return err;
 	}
 
+	/* Advertising is running */
```

Comment:
This one should follow the same pattern as `set_le_adv_enable_legacy`.
So should remove these lines and do it like this instead:
https://github.com/zephyrproject-rtos/zephyr/pull/24315/files#diff-8c3bce98d0cfacce4012133a2290e094R511-R512

Reply:
I didn't get the point here either. The link directs me to my changes

Reply:
I updated the link in both places.

Your total diff should look like this:
```
diff --git a/subsys/bluetooth/host/hci_core.c b/subsys/bluetooth/host/hci_core.c
index f787036003..052458921c 100644
--- a/subsys/bluetooth/host/hci_core.c
+++ b/subsys/bluetooth/host/hci_core.c
@@ -552,6 +552,7 @@ static int set_le_adv_enable_ext(const struct bt_le_ext_adv *adv,
                                 bool enable,
                                 const struct bt_le_ext_adv_start_param *param)
 {
+       struct cmd_state_set state;
        struct net_buf *buf;
        int err;
 
@@ -572,12 +573,14 @@ static int set_le_adv_enable_ext(const struct bt_le_ext_adv *adv,
        net_buf_add_le16(buf, param ? sys_cpu_to_le16(param->timeout) : 0);
        net_buf_add_u8(buf, param ? param->num_events : 0);
 
+       cmd_state_set_init(&state, bt_dev.flags, BT_DEV_ADVERTISING, enable);
+       cmd(buf)->state = &state;
+
        err = bt_hci_cmd_send_sync(BT_HCI_OP_LE_SET_EXT_ADV_ENABLE, buf, NULL);
        if (err) {
                return err;
        }
 
-       atomic_set_bit_to(bt_dev.flags, BT_DEV_ADVERTISING, enable);
 
        return 0;
 }
```